src/fields/field_set.hh

```diff
@@ -111,8 +111,10 @@ public:
 	 *
 	 * Input::Type::Record SomEquation::input_type=
 	 * 		Record("SomeEquation","equation's description")
-	 * 		.declare_key("data",Input::Type::Array(EqData().make_field_descriptor_type()),"List of field descriptors.");
+	 * 		.declare_key("data",Input::Type::Array(EqData().make_field_descriptor_type().close()),"List of field descriptors.");
```

Comment:
I.e. make_filed_descriptor returns unclosed record? That is potential source of errors, since close do the TypeRepository stuff. I propose to close the record and use copy constructor or copy_keys or something new to perform adding further keys ... but in completely new record.